src/runtime/cache.ts

```diff
@@ -67,14 +67,14 @@ export function defineCachedFunction<T = any>(
       ((await useStorage().getItem(cacheKey)) as any) || {};
 
     const ttl = (opts.maxAge ?? opts.maxAge ?? 0) * 1000;
-    if (ttl) {
+    if (ttl && !entry.expires) {
       entry.expires = Date.now() + ttl;
```

Comment:
We might need to create a local `expires` variable that takes `ttl` into account as well

Reply:
Would you mind sharing some more context? Still getting across the complexity of this function

Reply:
Hmm thinking more, in your example, you are generating an `expires` based on `Date.now()` + a certain ttl (from options or 60*60). Would it makes sense to allow directly modify ttl?

BTW to be honest it is a hacky workaround as you mentioned. validate is passing opts without expecting it to be modified.

Reply:
The feature is allowing you to dynamically set how long something can exist in the cache based on the entry value. Don't really mind how it's accomplished. 

Basically, the end user can modify the cache time of the svg, but we can only find out the cache time once it's been fetched initially.

Happy to rework this PR to introduce a feature to do it. Not sure the best name for it though, `onCacheSubmit(entry)` ?